### ISSUE TITLE
checking "Enable my custom filters" with no sources causes CPU usage to throttle

### ISSUE THREAD

**Issue by Tavanator** [uBlockOrigin/uBlock-issues#3481]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

When selecting "enable my custom filters" without any sources in the filter list, Firefox begins throttling CPU on websites.

### A specific URL where the issue occurs.

```yaml
*any website in which you visit has some slowness and stuttering
```

### Steps to Reproduce

1. Open UBlock extension
2. Select "settings" cogs at the bottom right of the window
3. navigate to "My Filters"
4. Check "enable my custom filters"
5. Do not add any sources in the list

### Expected behavior

The expected outcome should be performance issues within the browser, including the cursor disappearing when going over the browser, the webpage "lagging" behind when scrolling.

### Actual behavior

Web pages freeze for an unexpected amount of time before resuming.

### uBO version

uBlock Origin 1.61.2

### Browser name and version

Firefox 133.0 (64-bit)

### Operating System and version

Windows 11 version 23H2

**Comment by gwarser** [comment]
https://github.com/uBlockOrigin/uBlock-issues?tab=readme-ov-file#performance-issues

**Comment by gorhill** [comment]
> - [x]  using a new, unmodified browser profile.

That setting is checked by default in a new install, and there are no custom filters by default. What exact steps did you perform in that new profile to trigger the issue?

**Comment by Tavanator** [comment]
> > Do not add any sources in the list
> 
> > I added a source for testing but removed it.
> 
> What do you mean by "source"? The "My filters" pane is a place to add filters -- I don't understand what you mean by "add any sources".

A filter? Domain?

**Comment by gorhill** [comment]
Please confirm again: did you _really_ reproduce in a new browser profile with only uBO installed with no changes? If so, since the option _"Enable my custom filters"_ is enabled by default, how did you reproduce? You un-checked the option then checked again?

**Comment by Tavanator** [comment]
> Please confirm again: did you _really_ reproduce in a new browser profile with only uBO installed with no changes? If so, since the option _"Enable my custom filters"_ is enabled by default, how did you reproduce? You un-checked the option then checked again?

Correct. I added a filter/domain to the filters field, removed it and saw performance issues. Once I unchecked it and checked it again, the performance issues went away.

**Comment by Cowboy501** [comment]
> `[](url)`//
